### ISSUE TITLE
More than one self reference causes constraint issue with create_table_from_entity

### ISSUE THREAD

**Issue by Rudi3** [SeaQL/sea-orm#462]
**EDIT:** Possibly a duplicate of https://github.com/SeaQL/sea-orm/issues/405

If you have more than one linked (self referencing) relation like  https://www.sea-ql.org/SeaORM/docs/relation/self-referencing with the pg backend, the postgres constraint will have the same name and the table won't be created.

```
let db_postgres = DbBackend::Postgres;
let schema = Schema::new(db_postgres);

let stmt = &schema.create_table_from_entity(super::tree::Entity);

create_table(db, &stmt).await
```

```
#[derive(Clone, Debug, PartialEq, DeriveEntityModel, Deserialize, Serialize)]
#[sea_orm(table_name = "tree")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i32,
    pub parent_id: i32,
}

#[derive(Copy, Clone, Debug, EnumIter, DeriveRelation)]
pub enum Relation {
    #[sea_orm(
        belongs_to = "Entity",
        from = "Column::ParentId",
        to = "Column::Id"
    )]
    ParentNode,
    #[sea_orm(
        belongs_to = "Entity",
        from = "Column::Id",
        to = "Column::ParentId"
    )]
    Nodes
}

pub struct ParentNode;

impl Linked for ParentNode {
    type FromEntity = Entity;

    type ToEntity = Entity;

    fn link(&self) -> Vec<RelationDef> {
        vec![Relation::ParentNode.def()]
    }
}

pub struct Nodes;

impl Linked for Nodes {
    type FromEntity = Entity;

    type ToEntity = Entity;

    fn link(&self) -> Vec<RelationDef> {
        vec![Relation::Nodes.def()]
    }
}

```
```
CREATE TABLE "tree" (
  "id" serial NOT NULL PRIMARY KEY,
  "parent_id" integer NOT NULL,
  CONSTRAINT "fk-tree-tree" FOREIGN KEY ("parent_id") REFERENCES "tree" ("id"),
  CONSTRAINT "fk-tree-tree" FOREIGN KEY ("id") REFERENCES "tree" ("parent_id")
)
```

Removing the second constraint manually makes the code work. (However, I'm not sure if the inverse constraint should be created somehow in this case).

**Comment by Rudi3** [comment]
I suppose, for a tree-like structure, it should be something like this:
```
CREATE TABLE "tree" (
  "id" serial NOT NULL PRIMARY KEY,
  "parent_id" integer NOT NULL,
  UNIQUE ("parent_id", "id"),
  CONSTRAINT "fk-tree-tree" FOREIGN KEY ("parent_id", "id") REFERENCES "tree" ("parent_id", "id")
)
```

**Comment by billy1624** [comment]
Hey @Rudi3, thanks for the report! Yes, this is related to #405, and should be fixed on #417. You can try the PR :)

**Comment by billy1624** [comment]
Hey @Rudi3, I'll close this issue. Please reopen this if you still have problems :)
